tls/s2n_connection.c

```diff
@@ -741,11 +741,12 @@ int s2n_connection_get_client_cert_chain(struct s2n_connection *conn, uint8_t **
 int s2n_connection_get_cipher_preferences(struct s2n_connection *conn, const struct s2n_cipher_preferences **cipher_preferences)
 {
     notnull_check(conn);
+    notnull_check(conn->config);
     notnull_check(cipher_preferences);
 
     if (conn->security_policy_override != NULL) {
         *cipher_preferences = conn->security_policy_override->cipher_preferences;
-    } else {
+    } else if (conn->config->security_policy != NULL) {
```

Comment:
You probably want to make this a notnull_check instead, or you should add an else that throws an error. What happens if neither of these things are true?

Same for the others

Reply:
I could raise an error if neither is true.